packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -928,14 +933,17 @@ Future<T> showCupertinoModalPopup<T>({
   @required BuildContext context,
   @required WidgetBuilder builder,
   ImageFilter filter,
+  Color barrierColor = _kModalBarrierColor,
   bool useRootNavigator = true,
   bool semanticsDismissible,
 }) {
   assert(useRootNavigator != null);
+
   return Navigator.of(context, rootNavigator: useRootNavigator).push(
     _CupertinoModalPopupRoute<T>(
-      barrierColor: CupertinoDynamicColor.resolve(_kModalBarrierColor, context),
       barrierLabel: 'Dismiss',
+      barrierColor:
+          barrierColor?.alpha == 0 ? null : CupertinoDynamicColor.resolve(barrierColor, context),
```

Comment:
@xster pushed separate commit with this approach. Unfortunately, we need to check for `barrierColor` not to be `Color(0x00000000)` (I decided to keep this check as check for alpha channel being completely transparent) because of this `assert` in `_buildModalBarrier` https://github.com/flutter/flutter/blob/e92afc16b6445611b32fe19795768d16accd2c1d/packages/flutter/lib/src/widgets/routes.dart#L1408